cmd/ctr/commands/tasks/exec.go

```diff
@@ -94,7 +94,10 @@ var execCommand = cli.Command{
 		if err != nil {
 			return err
 		}
-		defer process.Delete(ctx)
+		// if detach, we should not call this defer
+		if !detach {
```

Comment:
Seems `statusC, err := process.Wait(ctx)` is also no need if exec process detached

Reply:
good comment, we can fix that in a follow up PR